app/com/github/pedrovgs/dotto/interpreter/led/ledController.scala

```diff
@@ -0,0 +1,74 @@
+package com.github.pedrovgs.dotto.interpreter.led
+
+import com.github.pedrovgs.dotto.core.algebra.dotto.{LedInteractions, MorseSentence}
+import com.github.pedrovgs.dotto.core.types._
+import com.pi4j.io.gpio._
+import play.api.Logger
+
+import scala.concurrent.duration._
+
+object ledController {
+
+  private val DotDuration = 0.1 seconds
+  private val DashDuration = DotDuration * 3
+  private val SpaceDuration = DashDuration * 3
+  private val SpaceBetweenSymbolsDuration = DotDuration * 2
+
+  private val SpaceBetweenSymbolsInteraction = LedInteraction(Low, SpaceBetweenSymbolsDuration)
+
+  /**
+    * Transforms a sequence of sequences of MorseSymbol instances into a sequence of led interactions. Every word
+    * inside the sentence represented with a sequence of MorseSymbols will be separated to the rest of the words using a
+    * space led interaction.
+    */
+  def toLedInteractions(morseSentence: MorseSentence): LedInteractions = {
+    morseSentence.flatten.map {
```

Comment:
map + flatten usually means flatMap. 2 flatMaps on the same type usually are simplified by a for comprehension. For comprehensions over collections act as generators meaning the left bound variable is each one of the items in the collection.